modules/swagger-parser-safe-url-resolver/src/main/java/io/swagger/parser/v3/urlresolver/PermittedUrlsChecker.java

```diff
@@ -0,0 +1,97 @@
+package io.swagger.parser.v3.urlresolver;
+
+import io.swagger.parser.v3.urlresolver.exceptions.HostDeniedException;
+import io.swagger.parser.v3.urlresolver.matchers.UrlPatternMatcher;
+import io.swagger.parser.v3.urlresolver.models.ResolvedUrl;
+import io.swagger.parser.v3.urlresolver.utils.NetUtils;
+
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+
+public class PermittedUrlsChecker {
+
+    private final UrlPatternMatcher allowlistMatcher;
+    private final UrlPatternMatcher denylistMatcher;
```

Comment:
maybe as `protected` to allow for extensibility?

Reply:
Absolutely, will update presently